### DISCUSSION TITLE
3.0.1 LDAP problem

### DISCUSSION THREAD

**Discussion by ognjen011** [netbox-community/netbox#7157] [General]
After successful  upgrade i enabled ldap auth. Then installed djnago ldap with pip3 install django-auth-ldap. But now when i browse to it seems the front end is broken. All i get is error below

<class 'AttributeError'>

'NoneType' object has no attribute '_base_manager'

Python version: 3.7.5
NetBox version: 3.0.1

I have been googling and found a similar issue here https://github.com/netbox-community/netbox/issues/5515  but it is closed without solution. If anyone has any ideas that would be great.

**Comment by ognjen011** [comment]
Fixed it with changing the file:
django/contrib/contenttypes/models.py on line 175 changing the line to something like this. No idea if this will break anything else. 

try: 
    return self.model_class()._base_manager.using(self._state.db).get(**kwargs)
except AttributeError: 
    return None

**Reply by jeremystretch** [reply]
FYI you've modified the Django code base, not NetBox. This is definitely not something to mess with; the fix will be in NetBox itself.

This is being tracked under issue #7149.

**Comment by ryanmerolle** [comment]
If that is indeed the issue, per the issue comment

> Found root cause.
> Seem relate to ldap configuration populate invalid email address.

I would suggest you go through the AD users being imported into netbox and see if there are any invalid email addresses and fix the AD user.  Then there is no issue with altering Django code base.

**Reply by ognjen011** [reply]
There is nothing  wrong with my openldap setup. I have been using it for years with netbox.

**Reply by ryanmerolle** [reply]
I was not suggesting that you open ldap was bad.  I was referring to the comment the poster mentioned from that bug you originally linked:

> Found root cause.
> Seem relate to ldap configuration populate invalid email address.
